### PR TITLE
Issue #85: Fix bug where detected sculk sensor clocks were broken but not dropped

### LINKED ISSUE #85: Sculk Snesors don't get detected
### Server Logs

not give any errors

### Expected behavior

During cycles made with the sculk sensor, the sculk sensor breaks and drop

### Actual behavior

During cycles made with the sculk sensor, the sculk sensor breaks but not drop

### Steps to reproduce

https://youtu.be/N2zdgaNpuMY

### Plugin and Datapack List

https://prnt.sc/SPR4nGKzrDQ6

not have datapack

### Paper version

https://prnt.sc/Z6bwfS6Oty2t

### Plugin version

https://prnt.sc/HQs7Phu9rm2U

### Other

I guess sculk sensor break only with silktouch  therefore not drop(forecast)

### PR DESCRIPTION
**Pull request by joelramilison**
## Proposed changes

Fixes #85 

Previously, whenever the plugin detected a clock using sculk sensors, it would destroy the sculk sensors but not drop them. This was because sculk sensors can only drop if you destroy them using a 'Silk Touch'-enchanted tool. Hence why I propose to set it when determining the dropped items in the getDrops() function.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/OneLiteFeatherNET/.github/blob/main/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java [hunk 1]
```diff
@@ -169,7 +171,9 @@ private void sendNotification(final Player player, final Location location) {
     private void breakBlock(@NotNull Location location) {
         Block block = location.getBlock();
         if (this.dropItems) {
-            var drops = block.getDrops();
+            ItemStack destroyTool = new ItemStack(Material.DIAMOND_PICKAXE);
```
Comment (TheMeinerLP) [inline src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java:174]: Don't re-create for each block again a new item stack. That call has the possibilities to create X new Objects in the Java Memory.

**Review by TheMeinerLP — CHANGES_REQUESTED** [review]

File: src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java (hunk 1 above)
Comment (joelramilison) [reply src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java]: New commit added

File: src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java [hunk 2]
```diff
@@ -169,7 +173,7 @@ private void sendNotification(final Player player, final Location location) {
     private void breakBlock(@NotNull Location location) {
         Block block = location.getBlock();
         if (this.dropItems) {
-            var drops = block.getDrops();
+            var drops = block.getDrops(SILK_TOUCH_PICKAXE);
```
Comment (TheMeinerLP) [inline src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java:176]: Can you double-check if the durability gets reduced ?

Comment (joelramilison) [reply src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java:176]: image
The method only checks what items **would** drop if the block were to be destroyed with a tool 👍🏻

Comment (TheMeinerLP) [reply src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java:176]: But using a tool consume durability from the tool

Comment (joelramilison) [reply src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java:176]: I've put it to the test and called the method 10,000 times, it still gave the same result.

`@Override
    public void onEnable() {
       ItemStack pickaxe = new ItemStack(Material.DIAMOND_PICKAXE);
       pickaxe.addEnchantment(Enchantment.SILK_TOUCH, 1);
       Location location = new Location(Bukkit.getWorlds().getFirst(), 0, 0, 0);
       Block block = location.getBlock();
       Collection<ItemStack> itemDrops = block.getDrops(pickaxe);
       int counter = 0;

       for (int i = 0; i < 10000; i++) {
           var newItemDrops = block.getDrops(pickaxe);
           if (!newItemDrops.equals(itemDrops)) {
               throw new RuntimeException("ERROR: Result of getDrops changed");
           }
           itemDrops = newItemDrops;
           counter++;
       }
       getLogger().info("Ran tool test " + counter + " times without error");
    }`

Comment (TheMeinerLP) [reply src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java:176]: Looks good then to me. An diamond pickaxe have just 1561 durability

**Review by TheMeinerLP — APPROVED** [review]

**Pull request merged**